AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -251,8 +252,9 @@ - (void)_initializeInstance
   [self _staticInitialize];
   _contentsScaleForDisplay = ASScreenScale();
   _displaySentinel = [[ASSentinel alloc] init];
+  _layoutRange = [[ASTranslationRange alloc] initWithLocation:0 length:0];
```

Comment:
Let's find a way to defer the allocation of this. After seeing that ASSentinel is one of the most commonly created and destroyed objects, along with the savings from removing common objects like ASLayoutOptions, it would be ideal to ensure we don't create these for situations that don't need them such as all leaf nodes at the bottom of a hierarchy.